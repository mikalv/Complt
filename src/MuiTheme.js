import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { pink500, pink700, tealA200 } from 'material-ui/styles/colors';

const MuiTheme = ({ children }) => (
  <MuiThemeProvider
    muiTheme={getMuiTheme({
      palette: {
        primary1Color: pink500,
        primary2Color: pink700,
        accent1Color: tealA200,
      },
    })}
  >
    {children}
  </MuiThemeProvider>
);

MuiTheme.propTypes = {
  children: React.PropTypes.node,
};

export default MuiTheme;
