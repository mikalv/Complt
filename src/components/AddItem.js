import React, { Component } from 'react';
import TextField from 'material-ui/TextField';
import ActionLabel from 'material-ui/svg-icons/action/label';
import ContentSend from 'material-ui/svg-icons/content/send';
import ActionAssignment from 'material-ui/svg-icons/action/assignment';
import ActionDone from 'material-ui/svg-icons/action/done';
import Paper from 'material-ui/Paper';
import IconButton from 'material-ui/IconButton';
import processItem from '../utils/processItem';
import './AddItem.css';

class AddItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      type: props.initialType || 'Task',
    };
    this.valueChange = this.valueChange.bind(this);
    this.ActionLabelTap = this.ActionLabelTap.bind(this);
    this.submitItem = this.submitItem.bind(this);
    this.switchType = this.switchType.bind(this);
  }
  valueChange(e) {
    this.setState({ value: e.target.value });
  }
  ActionLabelTap() {
    const value = this.state.value.trim();
    this.setState({ value: `${value} @` }, () => {
      this.valueInput.focus();
    });
  }
  submitItem(e) {
    e.preventDefault();
    const item = processItem(this.state.value, this.state.type);
    if (!item) return;
    this.setState({ value: '' });
    this.props.onAddItem(item);
  }
  switchType() {
    if (this.state.type === 'Project') {
      this.setState({ type: 'Task' });
    } else {
      this.setState({ type: 'Project' });
    }
  }
  render() {
    return (
      <form onSubmit={this.submitItem}>
        <Paper zDepth={2} style={{ padding: 10 }} className="md-drawer-relative">
          <div className="flex column">
            <TextField
              name="textInput"
              hintText={this.state.type === 'Project' ? 'e.g. Report' : 'e.g. Finish Report @work'}
              value={this.state.value}
              style={{ width: '100%' }}
              onChange={this.valueChange}
              ref={input => (this.valueInput = input)}
            />
            <div className="flex row space-between">
              <div>
                {this.state.type === 'Project' ? null : <IconButton disableTouchRipple onTouchTap={this.ActionLabelTap}>
                  <ActionLabel />
                </IconButton>}
              </div>
              <div>
                <IconButton type="submit" disableTouchRipple>
                  <ContentSend />
                </IconButton>
                {this.props.canChangeType ? <IconButton
                  disableTouchRipple
                  onTouchTap={this.switchType}
                >
                  {this.state.type === 'Project' ? <ActionDone /> : <ActionAssignment />}
                </IconButton> : null}
              </div>
            </div>
          </div>
        </Paper>
      </form>

    );
  }
}

AddItem.propTypes = {
  onAddItem: React.PropTypes.func.isRequired,
  initialType: React.PropTypes.oneOf(['Project', 'Task']),
  canChangeType: React.PropTypes.bool,
};

export default AddItem;
