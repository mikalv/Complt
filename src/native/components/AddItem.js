import React, { Component } from 'react';
import {
  View,
  StyleSheet,
  TextInput,
  KeyboardAvoidingView,
} from 'react-native';
import { IconToggle } from 'react-native-material-ui';
import colors from '../../common/colors';
import processItem from '../../common/utils/processItem';


class AddItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      isProject: props.initialIsProject,
    };
    this.valueChange = this.valueChange.bind(this);
    this.ActionLabelTap = this.ActionLabelTap.bind(this);
    this.submitItem = this.submitItem.bind(this);
    this.switchType = this.switchType.bind(this);
  }
  valueChange(value) {
    this.setState({ value });
  }
  ActionLabelTap() {
    const value = this.state.value.trim();
    this.setState({ value: `${value} @` });
  }
  submitItem() {
    const item = processItem(this.state.value, this.state.isProject);
    if (!item) return;
    this.setState({ value: '' });
    this.props.onAddItem(item);
  }
  switchType() {
    this.setState({ isProject: !this.state.isProject });
  }
  render() {
    return (
      <KeyboardAvoidingView behavior="padding" elevation={5} style={styles.container}>
        <View style={styles.flex}>
          <TextInput
            returnKeyType="send"
            style={styles.flex}
            underlineColorAndroid={colors.primary}
            value={this.state.value}
            onChangeText={this.valueChange}
            onSubmitEditing={this.submitItem}
            placeholder={this.state.isProject ? 'e.g. Report' : 'e.g. Finish Report @work'}
          />
        </View>
        <View style={styles.row}>
          <View style={styles.flex}>
            {!this.state.isProject ? <IconToggle name="label" onPress={this.ActionLabelTap} /> : null}
          </View>
          <View style={styles.flexEnd}>
            {this.props.canChangeType ? <IconToggle onPress={this.switchType} name={this.state.isProject ? 'assignment' : 'done'} /> : null}
            <IconToggle onPress={this.submitItem} name="send" />
          </View>
        </View>
      </KeyboardAvoidingView>);
  }
}

AddItem.propTypes = {
  onAddItem: React.PropTypes.func,
  initialIsProject: React.PropTypes.bool,
  canChangeType: React.PropTypes.bool,
};

const styles = StyleSheet.create({
  container: {
    height: 20,
    flex: 1,
    padding: 10,
    backgroundColor: '#fff',
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  flex: {
    flex: 1,
  },
  flexEnd: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
});

export default AddItem;
