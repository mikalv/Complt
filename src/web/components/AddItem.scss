@import "@material/elevation/mdc-elevation";
@import "../index";

.AddItem-fixed {
  position: fixed;
  bottom: 0;
  right: 0;
  width: 100%;
}
.AddItem {
  padding: 10px;
  background-color: $mdc-theme-background;
  @include mdc-elevation(24);
  margin-top: 5px;
}
.AddItem-input {
  width: 100%;
}
#add-item-input {
  width: 100%;
}
