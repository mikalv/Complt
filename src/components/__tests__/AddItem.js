import React from 'react';
import renderer from 'react-test-renderer';
import AddItem from '../AddItem';
import MuiTheme from '../../MuiTheme';

describe('AddItem component', () => {
  it('renders without crashing', () => {
    renderer.create(<MuiTheme>
      <AddItem />
    </MuiTheme>);
  });
});
