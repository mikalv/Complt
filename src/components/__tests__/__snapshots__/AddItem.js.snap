exports[`AddItem component renders correctly when canChangeType is true and initialIsProject is false 1`] = `
<AddItem
  canChangeType={true}
  onAddItem={[Function]}>
  <form
    onSubmit={[Function]}>
    <Paper
      className="md-drawer-relative"
      component="div"
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 10,
        }
      }
      zDepth={2}>
      <div
        className="md-paper md-paper--2 md-drawer-relative"
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": 10,
          }
        }>
        <div
          className="flex column">
          <TextField
            fullWidth={true}
            leftIconStateful={true}
            lineDirection="left"
            onChange={[Function]}
            passwordIconChildren="remove_red_eye"
            placeholder="e.g. Finish Report @work"
            rightIconStateful={true}
            type="text"
            value="">
            <div
              className="md-text-field-container md-full-width md-text-field-container--input"
              onClick={[Function]}
              style={
                Object {
                  "height": null,
                }
              }>
              <FloatingLabel
                active={false}
                error={false}
                floating={false}
                iconOffset={false} />
              <InputField
                fullWidth={true}
                inlineIndicator={false}
                onChange={[Function]}
                onFocus={[Function]}
                onHeightChange={[Function]}
                onKeyDown={[Function]}
                placeholder="e.g. Finish Report @work"
                type="text"
                value="">
                <input
                  className="md-text-field md-text md-full-width md-text-field--margin"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="e.g. Finish Report @work"
                  type="text"
                  value="" />
              </InputField>
              <TextFieldDivider
                active={false}
                error={false}
                lineDirection="left">
                <Divider
                  className="md-divider--text-field md-divider--expand-from-left">
                  <hr
                    className="md-divider md-divider--text-field md-divider--expand-from-left" />
                </Divider>
              </TextFieldDivider>
              <TextFieldMessage
                active={false}
                currentLength={0}
                error={false}
                leftIcon={false}
                rightIcon={false} />
            </div>
          </TextField>
          <div
            className="flex row space-between">
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                onClick={[Function]}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  onClick={[Function]}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          label
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                onClick={[Function]}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  onClick={[Function]}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          assignment
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                primary={true}
                type="submit">
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  primary={true}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}
                  type="submit">
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    primary={true}
                    type="submit">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-text--theme-primary md-ink--primary md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="submit">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          send
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
          </div>
        </div>
      </div>
    </Paper>
  </form>
</AddItem>
`;

exports[`AddItem component renders correctly when canChangeType is true and initialIsProject is true 1`] = `
<AddItem
  canChangeType={true}
  initialIsProject={true}
  onAddItem={[Function]}>
  <form
    onSubmit={[Function]}>
    <Paper
      className="md-drawer-relative"
      component="div"
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 10,
        }
      }
      zDepth={2}>
      <div
        className="md-paper md-paper--2 md-drawer-relative"
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": 10,
          }
        }>
        <div
          className="flex column">
          <TextField
            fullWidth={true}
            leftIconStateful={true}
            lineDirection="left"
            onChange={[Function]}
            passwordIconChildren="remove_red_eye"
            placeholder="e.g. Report"
            rightIconStateful={true}
            type="text"
            value="">
            <div
              className="md-text-field-container md-full-width md-text-field-container--input"
              onClick={[Function]}
              style={
                Object {
                  "height": null,
                }
              }>
              <FloatingLabel
                active={false}
                error={false}
                floating={false}
                iconOffset={false} />
              <InputField
                fullWidth={true}
                inlineIndicator={false}
                onChange={[Function]}
                onFocus={[Function]}
                onHeightChange={[Function]}
                onKeyDown={[Function]}
                placeholder="e.g. Report"
                type="text"
                value="">
                <input
                  className="md-text-field md-text md-full-width md-text-field--margin"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="e.g. Report"
                  type="text"
                  value="" />
              </InputField>
              <TextFieldDivider
                active={false}
                error={false}
                lineDirection="left">
                <Divider
                  className="md-divider--text-field md-divider--expand-from-left">
                  <hr
                    className="md-divider md-divider--text-field md-divider--expand-from-left" />
                </Divider>
              </TextFieldDivider>
              <TextFieldMessage
                active={false}
                currentLength={0}
                error={false}
                leftIcon={false}
                rightIcon={false} />
            </div>
          </TextField>
          <div
            className="flex row space-between">
            <div />
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                onClick={[Function]}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  onClick={[Function]}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          done
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                primary={true}
                type="submit">
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  primary={true}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}
                  type="submit">
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    primary={true}
                    type="submit">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-text--theme-primary md-ink--primary md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="submit">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          send
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
          </div>
        </div>
      </div>
    </Paper>
  </form>
</AddItem>
`;

exports[`AddItem component renders correctly when initialIsProject is false 1`] = `
<AddItem
  onAddItem={[Function]}>
  <form
    onSubmit={[Function]}>
    <Paper
      className="md-drawer-relative"
      component="div"
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 10,
        }
      }
      zDepth={2}>
      <div
        className="md-paper md-paper--2 md-drawer-relative"
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": 10,
          }
        }>
        <div
          className="flex column">
          <TextField
            fullWidth={true}
            leftIconStateful={true}
            lineDirection="left"
            onChange={[Function]}
            passwordIconChildren="remove_red_eye"
            placeholder="e.g. Finish Report @work"
            rightIconStateful={true}
            type="text"
            value="">
            <div
              className="md-text-field-container md-full-width md-text-field-container--input"
              onClick={[Function]}
              style={
                Object {
                  "height": null,
                }
              }>
              <FloatingLabel
                active={false}
                error={false}
                floating={false}
                iconOffset={false} />
              <InputField
                fullWidth={true}
                inlineIndicator={false}
                onChange={[Function]}
                onFocus={[Function]}
                onHeightChange={[Function]}
                onKeyDown={[Function]}
                placeholder="e.g. Finish Report @work"
                type="text"
                value="">
                <input
                  className="md-text-field md-text md-full-width md-text-field--margin"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="e.g. Finish Report @work"
                  type="text"
                  value="" />
              </InputField>
              <TextFieldDivider
                active={false}
                error={false}
                lineDirection="left">
                <Divider
                  className="md-divider--text-field md-divider--expand-from-left">
                  <hr
                    className="md-divider md-divider--text-field md-divider--expand-from-left" />
                </Divider>
              </TextFieldDivider>
              <TextFieldMessage
                active={false}
                currentLength={0}
                error={false}
                leftIcon={false}
                rightIcon={false} />
            </div>
          </TextField>
          <div
            className="flex row space-between">
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                onClick={[Function]}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  onClick={[Function]}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    onClick={[Function]}
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          label
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                primary={true}
                type="submit">
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  primary={true}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}
                  type="submit">
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    primary={true}
                    type="submit">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-text--theme-primary md-ink--primary md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="submit">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          send
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
          </div>
        </div>
      </div>
    </Paper>
  </form>
</AddItem>
`;

exports[`AddItem component renders correctly when initialIsProject is true 1`] = `
<AddItem
  initialIsProject={true}
  onAddItem={[Function]}>
  <form
    onSubmit={[Function]}>
    <Paper
      className="md-drawer-relative"
      component="div"
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 10,
        }
      }
      zDepth={2}>
      <div
        className="md-paper md-paper--2 md-drawer-relative"
        style={
          Object {
            "backgroundColor": "#fff",
            "padding": 10,
          }
        }>
        <div
          className="flex column">
          <TextField
            fullWidth={true}
            leftIconStateful={true}
            lineDirection="left"
            onChange={[Function]}
            passwordIconChildren="remove_red_eye"
            placeholder="e.g. Report"
            rightIconStateful={true}
            type="text"
            value="">
            <div
              className="md-text-field-container md-full-width md-text-field-container--input"
              onClick={[Function]}
              style={
                Object {
                  "height": null,
                }
              }>
              <FloatingLabel
                active={false}
                error={false}
                floating={false}
                iconOffset={false} />
              <InputField
                fullWidth={true}
                inlineIndicator={false}
                onChange={[Function]}
                onFocus={[Function]}
                onHeightChange={[Function]}
                onKeyDown={[Function]}
                placeholder="e.g. Report"
                type="text"
                value="">
                <input
                  className="md-text-field md-text md-full-width md-text-field--margin"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="e.g. Report"
                  type="text"
                  value="" />
              </InputField>
              <TextFieldDivider
                active={false}
                error={false}
                lineDirection="left">
                <Divider
                  className="md-divider--text-field md-divider--expand-from-left">
                  <hr
                    className="md-divider md-divider--text-field md-divider--expand-from-left" />
                </Divider>
              </TextFieldDivider>
              <TextFieldMessage
                active={false}
                currentLength={0}
                error={false}
                leftIcon={false}
                rightIcon={false} />
            </div>
          </TextField>
          <div
            className="flex row space-between">
            <div />
            <div>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}
                primary={true}
                type="submit">
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  primary={true}
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}
                  type="submit">
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    primary={true}
                    type="submit">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-text--theme-primary md-ink--primary md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="submit">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          send
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
          </div>
        </div>
      </div>
    </Paper>
  </form>
</AddItem>
`;
